scripts/tests/py/metadata.py

```diff
@@ -0,0 +1,160 @@
+# Copyright (c) 2023 Project CHIP Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from dataclasses import dataclass
+from pathlib import Path
+import re
+import yaml
+
+
+@dataclass
+class Metadata:
+    py_script_path: str = None
+    run: str = None
+    app: str = None
+    factoryreset: bool = False
+    comissioning_method: str = None
+    discriminator: int = None
+    kvs: str = None
+    storage_path: str = None
+    on_network_commission: str = None
+    passcode: int = None
+    trace_to: str = None
+    hex_arg: str = None
+    endpoint: str = None
+    manual_code: str = None
+    bool_arg: str = None
+    PICS: str = None
+    tests: str = None
+    int_arg: str = None
+    trace_to_appjson: str = None
+    trace_to_testjson: str = None
+    trace_toperfetto: str = None
+
+    def print(self):
+        print(repr(self))
+
+
+class Metadata_Reader:
+    # This class will be initialized with
+    # the name of the script folder and the environment
+    # file name
+    
+    def __init__(self, script_folder, env_file_name):
+        self.script_folder = script_folder
```

Comment:
I think this tries too much: it both reads and parses metadata and tries to be aware of a `script folder` that it recurses over.

lets split: make a reader that is only capable to read metadata one by one. leave the "multiple script parsing" and "aggregating all scripts" as a concern for some separate piece of code.